torch/package/package_importer.py

```diff
@@ -166,6 +167,8 @@ def load_pickle(self, package: str, resource: str, map_location=None) -> Any:
         """
         pickle_file = self._zipfile_path(package, resource)
         restore_location = _get_restore_location(map_location)
+        # to let TorchScript's reduce_packge read map location if needed
+        self.last_map_location = map_location
```

Comment:
I don't think storing this on the importer is a clean implementation, but I wasn't sure how else to pass the map_location passed in during the load_pickle call to the de-pickling function `reduce_package_script_module` in _script.py